### PR TITLE
test: transfer rollapp tokens between hub ↔ rollapp BEFORE rollapp has any finalized state on the hub

### LINKED ISSUE #187: rollapp-genesis: Transfer rollapp tokens between hub ↔ rollapp BEFORE rollapp has any finalized state on the hub
Test number 3 in [rollapp-genesis](https://github.com/dymensionxyz/e2e-tests/blob/main/tests_spec/rollapp_genesis.md)

### PR DESCRIPTION
**Pull request by ThanhNhann**
Close #187

### CODE REVIEW

File: tests/rollapp_genesis_event_test.go [hunk 1]
```diff
@@ -1024,3 +1024,413 @@ func TestHubTransferHubTriggerGenesis_EVM(t *testing.T) {
...
+	testutil.AssertBalance(t, ctx, rollapp1, rollappUserAddr, rollapp1.Config().Denom, walletAmount.Sub(transferRollAppToHubData.Amount))
+
+	// Get the IBC denom for urax on Hub
+	rollappTokenDenom := transfertypes.GetPrefixedDenom(channel.Counterparty.PortID, channel.Counterparty.ChannelID, rollapp1.Config().Denom)
+	rollappIBCDenom := transfertypes.ParseDenomTrace(rollappTokenDenom).IBCDenom()
+	
+	err = testutil.WaitForBlocks(ctx, 20, dymension, rollapp1)
+	require.NoError(t, err)
+
+	// Assert funds were returned to the sender after the timeout has occured
```
Comment (anhductn2001) [inline tests/rollapp_genesis_event_test.go:1195]: This comment confuses me

File: tests/rollapp_genesis_event_test.go [hunk 2]
```diff
@@ -1024,3 +1024,413 @@ func TestHubTransferHubTriggerGenesis_EVM(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// Assert balance was updated on the hub
+	testutil.AssertBalance(t, ctx, rollapp1, rollappUserAddr, rollapp1.Config().Denom, walletAmount.Sub(transferRollAppToHubData.Amount))
+
+	// Get the IBC denom for urax on Hub
+	rollappTokenDenom := transfertypes.GetPrefixedDenom(channel.Counterparty.PortID, channel.Counterparty.ChannelID, rollapp1.Config().Denom)
+	rollappIBCDenom := transfertypes.ParseDenomTrace(rollappTokenDenom).IBCDenom()
+	
+	err = testutil.WaitForBlocks(ctx, 20, dymension, rollapp1)
```
Comment (anhductn2001) [inline tests/rollapp_genesis_event_test.go:1192]: Should we wait for blocks or wait for state finalize

File: tests/rollapp_genesis_event_test.go [hunk 3]
```diff
@@ -1024,3 +1024,413 @@ func TestHubTransferHubTriggerGenesis_EVM(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// Assert funds were returned to the sender after the timeout has occured
```
Comment (anhductn2001) [inline tests/rollapp_genesis_event_test.go:1400]: same above

File: tests/rollapp_genesis_event_test.go [hunk 4]
```diff
@@ -1024,3 +1024,413 @@ func TestHubTransferHubTriggerGenesis_EVM(t *testing.T) {
...
+	require.NoError(t, err)
+
+	// Assert balance was updated on the hub
+	testutil.AssertBalance(t, ctx, rollapp1, rollappUserAddr, rollapp1.Config().Denom, walletAmount.Sub(transferRollAppToHubData.Amount))
+
+	// Get the IBC denom for urax on Hub
+	rollappTokenDenom := transfertypes.GetPrefixedDenom(channel.Counterparty.PortID, channel.Counterparty.ChannelID, rollapp1.Config().Denom)
+	rollappIBCDenom := transfertypes.ParseDenomTrace(rollappTokenDenom).IBCDenom()
+	
+	err = testutil.WaitForBlocks(ctx, 20, dymension, rollapp1)
```
Comment (anhductn2001) [inline tests/rollapp_genesis_event_test.go:1397]: same above

File: tests/rollapp_genesis_event_test.go (hunk 2 above)
Comment (ThanhNhann) [reply tests/rollapp_genesis_event_test.go:1192]: this is our requirement for this test: `transfer rollapp tokens between hub ↔ rollapp BEFORE rollapp has any finalized state on the hub`

File: tests/rollapp_genesis_event_test.go (hunk 4 above)
Comment (ThanhNhann) [reply tests/rollapp_genesis_event_test.go]: I updated the logic for checking finalize @anhductn2001

File: tests/rollapp_genesis_event_test.go [hunk 5]
```diff
@@ -1115,89 +1115,81 @@ func TestTransferTriggerGenesisBoth_EVM(t *testing.T) {
...
 	// trigger genesis event for hub and rollapp
 	triggerHubGenesisEvent(t, dymension, rollapp_params)
 
-	registerGenesisEventTriggerer(t, rollapp1.CosmosChain, rollappUser.KeyName(), rollappUserAddr, "hubgenesis", "GenesisTriggererAllowlist")
-	_, err = rollapp1.Validators[0].ExecTx(ctx, rollappUserAddr, "hubgenesis", "genesis-event", dymension.GetChainID(), channel.ChannelID)
+	registerGenesisEventTriggerer(t, rollapp1.CosmosChain, rollappUser1.KeyName(), rollappUserAddr1, "hubgenesis", "GenesisTriggererAllowlist")
+	_, err = rollapp1.Validators[0].ExecTx(ctx, rollappUserAddr1, "hubgenesis", "genesis-event", dymension.GetChainID(), channel.ChannelID)
 	require.NoError(t, err)
 
 	err = r.StartRelayer(ctx, eRep, ibcPath)
```
Comment (anhductn2001) [inline tests/rollapp_genesis_event_test.go:1152]: We should startRelayer after the last tx be sent.

**Pull request merged**